File: leaf-system/src/main/java/com/leaf/domain/LeafUserDetail.java
package com.leaf.domain;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

/**
 * @author jhlz
 * @version 1.0.0
 */
@Entity
@Table(name = "leaf_user_detail")
public interface LeafUserDetail extends ID {

    /**
     * 姓氏
     */
    @Key
    String firstName();

    /**
     * 名称
     */
    @Key
    String lastName();

    /**
     * 民族
     */
    @Nullable
    String ethnic();

    /**
     * 性别
     */
    @Nullable
    String gender();

    /**
     * 年龄
     */
    @Nullable
    Integer age();

    /**
     * 现住址
     */
    @Nullable
    String address();

    /**
     * 故乡
     */
    @Nullable
    String hometown();

    /**
     * 身份证号
     */
    @Nullable
    String idCard();

    // @OneToOne
    // // @JoinColumn(name = "id")
    // LeafUser user();
}

File: src/main/kotlin/com/jhlz/domain/LeafUserDetail.kt
package com.jhlz.domain

import org.babyfish.jimmer.sql.*

/**
 * @author jhlz
 * @version 1.0.0
 */
@Entity
@Table(name = "leaf_user_detail")
interface LeafUserDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    /**
     * 姓氏
     */
    @Key
    val firstName: String?

    /**
     * 名称
     */
    @Key
    val lastName: String?

    /**
     * 民族
     */
    val ethnic: String?

    /**
     * 性别
     */
    val gender: String?

    /**
     * 年龄
     */
    val age: Int?

    /**
     * 现住址
     */
    val address: String?

    /**
     * 故乡
     */
    val hometown: String?

    /**
     * 身份证号
     */
    val idCard: String?

}
